### PR TITLE
Draft for simple polygon renderer

### PR DESCRIPTION
**Pull request by kb173**
Adds the ability to render simple polygons rather than only the specialized building polygons we have at the moment.

Still WIP (unclean code and doesn't look quite as intended) but working in principle.

@MathiasBaumgartinger would be great if you could have a look at the changes to the Layer and particularly SQL loading changes

### CODE REVIEW

File: Layers/Layer.gd [hunk 1]
```diff
@@ -25,6 +25,7 @@ enum RenderType {
 	PATH,
 	CONNECTED_OBJECT,
 	POLYGON,
+	GRADIENT_POLYGON,
```
Comment (MathiasBaumgartinger) [inline Layers/Layer.gd:28]: Is this really necessary? For the buildings it was not necessary to create a new enum type.

File: Layers/LayerConfigurator.gd [hunk 2]
```diff
@@ -467,6 +458,9 @@ func load_polygon_layer(db, layer_config, geo_layers_config, extended_as: Layer.
 	if get_extension_by_key(db, "extends_as", layer_config.id) == "BuildingRenderInfo":
 		if extended_as == null:
 			return load_buildings(db, layer_config, geo_layers_config)
+	elif get_extension_by_key(db, "extends_as", layer_config.id) == "GradientPolygonRenderInfo":
```
Comment (MathiasBaumgartinger) [inline Layers/LayerConfigurator.gd:461]: Shouldn't this just be the default case for rendering polygons? I think we do not have a default for rendering polygons as of now so this would be a quite good fit, would it not?

File: Layers/LayerRenderersConfig.gd [hunk 3]
```diff
@@ -5,6 +5,7 @@ var renderers = {
 	Layer.RenderType.BASIC_TERRAIN: preload("res://Layers/Renderers/Terrain/BasicTerrainRenderer.tscn"),
 	Layer.RenderType.REALISTIC_TERRAIN: preload("res://Layers/Renderers/Terrain/RealisticTerrainRenderer.tscn"),
 	Layer.RenderType.POLYGON: preload("res://Layers/Renderers/Polygon/PolygonRenderer.tscn"),
+	Layer.RenderType.GRADIENT_POLYGON: preload("res://Layers/Renderers/Polygon/GradientPolygonRenderer.tscn"),
```
Comment (MathiasBaumgartinger) [inline Layers/LayerRenderersConfig.gd:8]: Do we need a new scene or is it possible to generically put both render-types into one? In the``GradientPolygonRenderer`` it appears that it shares a lot of code with the ``PolygonRenderer`` (which should not be named like that anyways in my opinion)

File: Layers/Renderers/Terrain/BasicTerrainRenderer.tscn [hunk 4]
```diff
@@ -29,5 +29,5 @@ script = ExtResource( 4 )
 
 [node name="TerrainLOD0" parent="." index="0" instance=ExtResource( 2 )]
 material_override = SubResource( 1 )
-size = 25000.0
-ortho_resolution = 10000
+size = 10000.0
+ortho_resolution = 5000
```
Comment (MathiasBaumgartinger) [inline Layers/Renderers/Terrain/BasicTerrainRenderer.tscn:33]: Is this change necessary?

File: Layers/Renderers/Terrain/TerrainLOD.gd [hunk 5]
```diff
@@ -94,47 +94,48 @@ func build():
...
+					current_specular_ground_textures = Vegetation.get_ground_sheet_texture(group_array, "specular")
+					current_ambient_ground_textures = Vegetation.get_ground_sheet_texture(group_array, "ambient")
+					current_roughness_ground_textures = Vegetation.get_ground_sheet_texture(group_array, "roughness")
+				
+				if load_fade_textures:
+					current_albedo_fade_textures = Vegetation.get_fade_sheet_texture(group_array, "albedo")
+					current_normal_fade_textures = Vegetation.get_fade_sheet_texture(group_array, "normal")
 
 func apply_textures():
 	rebuild_aabb()
```
Comment (MathiasBaumgartinger) [inline Layers/Renderers/Terrain/TerrainLOD.gd:141]: Don't really understand what is going on here, but i guess it is not breaking :D

**Review by MathiasBaumgartinger — CHANGES_REQUESTED** [review]

File: Layers/LayerConfigurator.gd (hunk 2 above)
Comment (kb173) [reply Layers/LayerConfigurator.gd:461]: I think this is tricky: it's true that it would be a sensible default case, but then again, there are variables needed here (at least a height and a color) which the buildings do not need, since they are given attribute names for those values. How would you resolve that?

File: Layers/LayerRenderersConfig.gd (hunk 3 above)
Comment (kb173) [reply Layers/LayerRenderersConfig.gd:8]: Yeah I've been meaning to rename the PolygonRenderer to BuildingRenderer. I think it'd be good to separate them (perhaps with BuildingRenderer inheriting from PolygonRenderer) since the buildings have a lot of very specific logic.

File: Layers/Renderers/Terrain/TerrainLOD.gd (hunk 5 above)
Comment (kb173) [reply Layers/Renderers/Terrain/TerrainLOD.gd:141]: I think that was a fix to get the BasicTerrainRenderer to work again. I'll check that and cherry-pick those changes to main if that's the case

**Comment by MathiasBaumgartinger** [comment]
Uses deprecated SQL functionalities. Closing this without merging
